src/services/swap_service.ts

```diff
@@ -219,7 +221,39 @@ export class SwapService {
         const nativeFills = _.flatten(swapQuote.orders.map(order => order.fills)).filter(
             fill => fill.source === ERC20BridgeSource.Native,
         );
-        adjustedWorstCaseProtocolFee = new BigNumber(PROTOCOL_FEE_MULTIPLIER).times(gasPrice).times(nativeFills.length);
+
+        const hasRFQTOrders = swapQuote.orders.some(order => {
+            const isNativeOrder = order.fills.some(fill => fill.source === ERC20BridgeSource.Native);
+            const hasTakerAddress = order.takerAddress !== NULL_ADDRESS;
+
+            return isNativeOrder && hasTakerAddress;
+        });
+
+        // NOTE: Takers have a tendency to bump the gas price in order to speed up their trades
+        // for Native orders this often leads to an insufficient protocol fee being included and
+        // potential RFQT orders cannot be filled, so to avoid this we pad the protocol fee amount
+        // specifically when RFQT orders are included in the quote
+        if (hasRFQTOrders) {
+            const maxGasPricePadding = gasPrice.times(RFQT_PROTOCOL_FEE_GAS_PRICE_MAX_PADDING_MULTIPLIER);
+            const fastestGasPrice = await ethGasStationUtils.getGasPriceOrThrowAsync('fastest').catch(err => {
```

Comment:
This potential extra gas price lookup (if there's nothing in cache) adds some extra response time for RFQT orders but given that this is a short-term bandaid I think that is okay. Wdyt?

Reply:
It's ok since its an internal proxy cache and should be fast. And in case it fails it is handled well.